Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -0,0 +1,648 @@
+    // rather than the game mode as a whole, so please use the singular form. Official translations:
+    // 名作トライアル (Japanese), Masterpieces (English), Meisterstücke (German), Chefs-d'œuvre
+    // (French), Clásicos (Spanish), Capolavori (Italian), 클래식 게임 체험판 (Korean).
+    // If your language is not one of the languages above, consider leaving the string untranslated
+    // so that people will recognize it as the name of the game mode.
+    name = StringFromFormat(GetStringT("%s (Masterpiece)").c_str(), name.c_str());
+  }
+
+  if (partition.offset % VolumeWii::BLOCK_TOTAL_SIZE != 0)
+  {
+    AddProblem(Severity::Medium,
+               StringFromFormat(GetStringT("The %s partition is not properly aligned.").c_str(),
+                                name.c_str()));
+  }
+
+  CheckCorrectlySigned(
+      partition, StringFromFormat(GetStringT("The %s partition is not correctly signed.").c_str(),
+                                  name.c_str()));
+
+  bool invalid_disc_header = false;
+  std::vector<u8> disc_header(0x80);
+  if (!m_volume.Read(0, disc_header.size(), disc_header.data(), partition))
+  {
+    invalid_disc_header = true;
+  }
+  else if (Common::swap32(disc_header.data() + 0x18) != 0x5D1C9EA3)
+  {
+    for (size_t i = 0; i < disc_header.size(); i += 4)
+    {
+      if (Common::swap32(disc_header.data() + i) != i)
+      {
+        invalid_disc_header = true;
+        break;
+      }
+    }
+
+    // The loop above ends without breaking for discs that legitimately lack updates.
+    // No such discs have been released to end users. Most such discs are debug signed,
+    // but there is apparently at least one that is retail signed, the Movie-Ch Install Disc.
+    return false;
+  }
+  if (invalid_disc_header)
+  {
+    // This can happen when certain programs that create WBFS files scrub the entirety of
+    // the Masterpiece partitions in Super Smash Bros. Brawl without removing them from
+    // the partition table. https://bugs.dolphin-emu.org/issues/8733
+    const std::string text = StringFromFormat(
+        GetStringT("The %s partition does not seem to contain valid data.").c_str(), name.c_str());
+    AddProblem(severity, text);
+    return false;
+  }
+
+  const DiscIO::FileSystem* filesystem = m_volume.GetFileSystem(partition);
+  if (!filesystem)
+  {
+    const std::string text = StringFromFormat(
+        GetStringT("The %s partition does not have a valid file system.").c_str(), name.c_str());
+    AddProblem(severity, text);
+    return false;
+  }
+
+  if (type == PARTITION_UPDATE)
+  {
+    std::unique_ptr<FileInfo> file_info = filesystem->FindFileInfo("_sys");
+    bool has_correct_ios = false;
+    if (file_info)
+    {
+      const IOS::ES::TMDReader& tmd = m_volume.GetTMD(m_volume.GetGamePartition());
+      if (tmd.IsValid())
+      {
+        const std::string correct_ios = "IOS" + std::to_string(tmd.GetIOSId() & 0xFF) + "-";
```

Comment:
Note to self: Dash at the end to avoid partial matches. File names are "IOS‌ _nn_-_xx_-v _vv_.wad"

Reply:
By the way, do you know what the *xx* stands for? It seems to always be 64, but I'm not sure why.

Reply:
I'm afraid not. Maybe 64-bit? No idea.